src/Cardano/Wallet.hs

```diff
@@ -117,6 +132,16 @@ mkWalletLayer db network = WalletLayer
             return (w, error "FIXME: store and retrieve wallet metadata")
 
     , watchWallet = liftIO . listen network . applyBlocks
+    , signTx = \wid ins outs -> liftIO (readCheckpoint db (PrimaryKey wid)) >>= \case
+        Nothing ->
+            undefined
+        Just w -> do
+            let state = getState w
+            let rootXPrv = error "TODO"
+            let password = error "TODO"
```

Comment:
Password would have to be passed as a parameter here, we can't look it up because we don't store it :+1: